### PR TITLE
Commerce 7829

### PR DESCRIPTION
**Pull request by andrea-ale-sbarra**
### Context

[JIRA Ticket](https://issues.liferay.com/browse/COMMERCE-7829)

To discontinue a product means just mark the sku as discontinued. The scope of this new feature is to let the buyer purchase the product regardless of the item that is actually added to cart. In this use case scenario the admin user will configure the product so that it will be marked as discontinued and if a replacement item is configured, the buyer can still purchase the product via its replacement.

### Based on https://github.com/liferay-commerce/liferay-portal/pull/2046

### CODE REVIEW

**Comment by andrea-ale-sbarra** [comment]
ci:test:sf

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by andrea-ale-sbarra** [comment]
@marco-leo see 40d0040.. 73946bc

File: modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java [hunk 1]
```diff
@@ -117,6 +111,21 @@ public Sku toDTO(DTOConverterContext dtoConverterContext) throws Exception {
...
+	private void _setSkuReplacement(
+			Sku sku, long replacementCProductId,
+			String replacementCPInstanceUuid)
+		throws Exception {
+
+		CPInstance replacementCPInstance =
+			_cpInstanceService.fetchCProductInstance(
+				replacementCProductId, replacementCPInstanceUuid);
+
+		sku.setDiscontinuedDate(replacementCPInstance.getDiscontinuedDate());
```
Comment (alessiorendina) [inline modules/apps/commerce/headless/headless-commerce/headless-commerce-admin-catalog-impl/src/main/java/com/liferay/headless/commerce/admin/catalog/internal/dto/v1_0/converter/SkuDTOConverter.java:123]: We can do these setters with lambda functions instead and return the object (Sku) directly without creating a variable

**Review by alessiorendina — CHANGES_REQUESTED** [review]
@andrea-ale-sbarra take a look at the comment, thanks.

**Comment by andrea-ale-sbarra** [comment]
ci:forward

**Pull request closed without merging**